src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py

```diff
@@ -382,7 +384,16 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
 register_cli_argument('network traffic-manager profile check-dns', 'type', help=argparse.SUPPRESS, default='Microsoft.Network/trafficManagerProfiles')
 
 # Traffic manager endpoints
-register_cli_argument('network traffic-manager endpoint', 'endpoint_name', name_arg_type, id_part='name', help='Endpoint name.', completer=get_tm_endpoint_completion_list())
 endpoint_types = ['azureEndpoints', 'externalEndpoints', 'nestedEndpoints']
+register_cli_argument('network traffic-manager endpoint', 'endpoint_name', name_arg_type, id_part='name', help='Endpoint name.', completer=get_tm_endpoint_completion_list())
 register_cli_argument('network traffic-manager endpoint', 'endpoint_type', options_list=('--type',), help='Endpoint type.  Values include: {}.'.format(', '.join(endpoint_types)), completer=get_generic_completion_list(endpoint_types))
 register_cli_argument('network traffic-manager endpoint', 'profile_name', help='Name of parent profile.', completer=get_resource_name_completion_list('Microsoft.Network/trafficManagerProfiles'))
+
+# DNS
+register_cli_argument('network dns zone', 'zone_name', name_arg_type)
+register_cli_argument('network dns record-set', 'record_set_name', name_arg_type, help='The name of the RecordSet, relative to the name of the zone.')
+register_cli_argument('network dns record-set', 'relative_record_set_name', name_arg_type, help='The name of the RecordSet, relative to the name of the zone.')
+register_cli_argument('network dns record-set', 'zone_name', help='The name of the zone without a terminating dot.')
+register_cli_argument('network dns record-set create', 'record_set_type', options_list=('--type',), choices=get_enum_choices(RecordType), completer=get_enum_type_completion_list(RecordType), type=str.upper)
+register_cli_argument('network dns record-set', 'record_type', options_list=('--type',), choices=get_enum_choices(RecordType), completer=get_enum_type_completion_list(RecordType), type=str.upper)
+register_cli_argument('network dns record-set record', 'record_set_name', options_list=('--record-set-name',))
```

Comment:
there is currently no way to list record sets, I have an issue open and an email out to the sdk team

---
In reply to: [75556541](https://github.com/Azure/azure-cli/pull/718#discussion_r75556541) [](ancestors = 75556541)